step/step.go

```diff
@@ -0,0 +1,553 @@
+package step
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/bitrise-io/go-steputils/stepconf"
+	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-io/go-utils/progress"
+	"github.com/bitrise-io/go-utils/retry"
+	"github.com/bitrise-steplib/steps-xcode-test/cache"
+	"github.com/bitrise-steplib/steps-xcode-test/output"
+	"github.com/bitrise-steplib/steps-xcode-test/simulator"
+	"github.com/bitrise-steplib/steps-xcode-test/xcodebuild"
+	"github.com/bitrise-steplib/steps-xcode-test/xcpretty"
+)
+
+const (
+	minSupportedXcodeMajorVersion = 6
+)
+
+const simulatorShutdownState = "Shutdown"
+
+// Output tools ...
+const (
+	XcodebuildTool = "xcodebuild"
+	XcprettyTool   = "xcpretty"
+)
+
+const (
+	testRepetitionModeNone = "none"
+)
+
+// Input ...
+type Input struct {
+	// Project Parameters
+	ProjectPath string `env:"project_path,required"`
+	Scheme      string `env:"scheme,required"`
+	TestPlan    string `env:"test_plan"`
+
+	// Simulator Configs
+	SimulatorPlatform  string `env:"simulator_platform,required"`
+	SimulatorDevice    string `env:"simulator_device,required"`
+	SimulatorOsVersion string `env:"simulator_os_version,required"`
+
+	// Test Repetition
+	TestRepetitionMode             string `env:"test_repetition_mode,opt[none,until_failure,retry_on_failure,up_until_maximum_repetitions]"`
+	MaximumTestRepetitions         int    `env:"maximum_test_repetitions,required"`
+	RelaunchTestsForEachRepetition bool   `env:"relaunch_tests_for_each_repetition,opt[yes,no]"`
+
+	// Test Run Configs
+	OutputTool            string `env:"output_tool,opt[xcpretty,xcodebuild]"`
+	IsCleanBuild          bool   `env:"is_clean_build,opt[yes,no]"`
+	IsSingleBuild         bool   `env:"single_build,opt[true,false]"`
+	ShouldBuildBeforeTest bool   `env:"should_build_before_test,opt[yes,no]"`
+
+	RetryTestsOnFailure       bool `env:"should_retry_test_on_fail,opt[yes,no]"`
+	DisableIndexWhileBuilding bool `env:"disable_index_while_building,opt[yes,no]"`
+	GenerateCodeCoverageFiles bool `env:"generate_code_coverage_files,opt[yes,no]"`
+	HeadlessMode              bool `env:"headless_mode,opt[yes,no]"`
+
+	TestOptions         string `env:"xcodebuild_test_options"`
+	XcprettyTestOptions string `env:"xcpretty_test_options"`
+
+	// Debug
+	Verbose                     bool   `env:"verbose,opt[yes,no]"`
+	CollectSimulatorDiagnostics string `env:"collect_simulator_diagnostics,opt[always,on_failure,never]"`
+
+	// Output export
+	DeployDir             string `env:"BITRISE_DEPLOY_DIR"`
+	ExportUITestArtifacts bool   `env:"export_uitest_artifacts,opt[true,false]"`
+
+	CacheLevel string `env:"cache_level,opt[none,swift_packages]"`
+}
+
+type exportCondition int
```

Comment:
We don't necessarily have to change this now, just interested in your opinion: would it be better to use a `string` enum instead?

Reply:
One of the upsides would be that we wouldn't need the extra parser method. Otherwise, they pretty much have the same pros and cons. Wdyt?

Reply:
Switched to string enum for `exportCondition` as it simplified the ProcessConfig. 
Manually parsing and validating the related step input was unnecessary as `stepconf` package with the applied struct field tags makes sure the input value is one f the supported values.